pkg/net/net.go

```diff
@@ -111,3 +112,45 @@ func PrefixesOverlap(cidr1 net.IPNet, cidr2 net.IPNet) bool {
 	}
 	return false
 }
+
+func GetNetworkInterface() (string, error) {
```

Comment:
Out of curiosity: how is this used? If multiple physical interfaces are connected to the host, GetNetworkInterface will return one of the random interfaces that match the constraint. But from one run to another on the same multi-nic host, the interface returned by this function will differ. Is this acceptable? If not, did you consider using the approach used in GetIPAddress() (which is designed to always return the public facing IP - which should be consistent over time)?
Also, as a nitpick, I think we should name this function accordingly (GetNetworkInterface() isn't clear about what interface is being picked).

Reply:
Thanks for the help!

The background for this is that to replace Failover Cluster, I'm trying to move a static ip address from one cloud agent to another based on who is the current winner of an election.  To do that, I'm using the PS New-NetIPAddress cmdlet. 

This is method is used when a configuration is not provided, so it is a fallback.  I would be thrilled to have a better (more predictable implementation), but at the time I did not know of one.  Is there a better way to grab the public facing IP address' network interface?

Reply:
Thanks for clarifying! This helps!

So, yes, there is a way to guarantee the interface is always the same. Typically we do that by looking at the route configured on the host (routes are stable over time, unless the admin changes them, which is unlikely). The 0.0.0.0/0 route with the smallest metric is considered the primary public interface. In order to simplify our life, what we typically do in golang is to "Dial" an address on the internet and look at what the local IP address is, then we look for the IP address in the list of all the interfaces. IOW, we let the TCP/IP stack do the route resolution for us. 
The "Dialing" step is implemented in GetIPAddress() in this file. GetIPAddress() will return the local IP of the public facing interface. Once you have it, it should normally be possible to loop through all the network interfaces and find the one that contains that IP (through https://golang.org/pkg/net/#Interface.Addrs).

Hope this helps!

Reply:
awesome ... thanks!  i tried to do as you suggested and have pushed an update.

Reply:
Awesome! Thanks a lot!

Reply:
Nitpicky note: the address returned by GetIPAddress is already guaranteed to be coming from an interface that is up and that is not the loopback. So, the code could be simplified. No biggie though, I am nitpicking.

Reply:
That makes sense and would definitely simplicy the code.  Would the comparison of IsLoopback would be quicker than comparing 127.0.0.1 to the GetIPAddress result?